crates/rpc-types-engine/src/envelope.rs

```diff
@@ -268,6 +326,38 @@ impl OpNetworkPayloadEnvelope {
             parent_beacon_block_root: Some(parent_beacon_block_root),
         })
     }
+
+    /// Encodes a payload envelope as a snappy-compressed byte array.
+    #[cfg(feature = "std")]
+    pub fn encode_v4(&self) -> Result<Vec<u8>, PayloadEnvelopeEncodeError> {
+        use ssz::Encode;
+        let execution_payload_v4 = match &self.payload {
+            OpExecutionPayload::V4(execution_payload_v4) => execution_payload_v4,
+            _ => return Err(PayloadEnvelopeEncodeError::WrongVersion),
+        };
+
+        let mut data = Vec::new();
+        let mut sig = self.signature.as_bytes();
+        sig[64] = self.signature.v() as u8;
```

Comment:
Ah nice. I didn't notice we aliased the import of the deprecated `Signature` type -> `PrimitiveSignature` in my prev comment. This works too with the new type 😄